torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -1244,9 +1269,12 @@ class TORCH_CUDA_CU_API ForLoop final : public Expr {
 
  private:
   Val* const index_ = nullptr;
+  Val* const extent_ = nullptr;
```

Comment:
Should `extent_` always be used instead of `inter_domain_->extent()`? I see both are used and don't know that's intentional.

Reply:
`extent_` is primarily used in the codegen. `for(int idx = init; idx < extent; idx += offset)`

Reply:
Even in codegen.cpp, `ForLoop::iter_domain()->rawExtent()` is used. I think it's confusing to have both `ForLoop::extent()` and `ForLoop::iter_domain()->extent()`.

I see the only place where `ForLoop::extent()` can be different from `ForLoop::iter_domain()->extent()` is `cloneVectorizedLoopNests`. I wonder if it would make more sense to create a new `kir::IterDomain` rather than reusing the same `kir::IterDomain` for the cloned loops. That way, we don't need to have `ForLoop::extent()` to override the extent of `ForLoop::iter_domain()`.

Reply:
@csarofeen Can we create an new iterDomain for the different loops?

I forgot if this idea was thrown around before.

Reply:
Please review #784

Reply:
Seemed reasonable but left one comment on the PR.